packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -412,11 +418,58 @@ export class VueElement extends BaseClass {
         const s = document.createElement('style')
         s.textContent = css
         this.shadowRoot!.appendChild(s)
+        this._childStylesAnchor = s
+        // record for HMR
+        if (__DEV__) {
+          ;(this._styles || (this._styles = [])).push(s)
+        }
+      })
+    }
+  }
+
+  // The method used by custom element child components
+  // to add styles to the shadow dom
+  protected _addChildStyles(
+    styles: string[] | undefined,
+    instance: ComponentInternalInstance
+  ) {
+    if (styles) {
+      styles.forEach((css, index) => {
+        const s = document.createElement('style')
+        s.textContent = css
+        // Generate ids and record them, and delete style tags based on
+        // them when components are unmounted
+        const ceStyleId = `data-v-ce-${new Date().getTime()}`
+        ;(instance.cecStyleIds || (instance.cecStyleIds = [])).push(ceStyleId)
+        s.setAttribute(ceStyleId, '')
+
+        if (this._childStylesAnchor) {
+          this.shadowRoot!.insertBefore(
+            s,
+            (this._childStylesAnchor as Node).nextSibling
+          )
+        } else {
+          this.shadowRoot!.appendChild(s)
+        }
```

Comment:
Do you mean directly storing the dom of the style tag? I don't recommend doing this, so a large number of dom references will be stored on the component instance. In fact, this is a big consumption in memory, and it may cause memory overflow, so I used `ceStyleId` at that time

Reply:
I use set and component uid to mark, so that I only record a cecStyleId in the component dimension instead of marking each style tag, so the memory usage can be further reduced. In addition, there was a problem with using getTime before. The value returned by getTime is not unique in a very short time